concent_api/core/migrations/0006_add_and_update_deadline_and_size.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-06-19 09:37
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db import models
+from common.helpers import deserialize_message
+from common.helpers import parse_timestamp_to_utc_datetime
+
+
+def populate_subtask_size_field(apps, _schema_editor):
+    Subtask = apps.get_model('core', 'Subtask')
+    for subtask in Subtask.objects.filter(result_package_size__isnull=True):
+        subtask.result_package_size = deserialize_message(
+            subtask.report_computed_task.data
+        ).size
+        subtask.full_clean()
+        subtask.save()
+
+
+def populate_subtask_deadline_field(apps, _schema_editor):
+    Subtask = apps.get_model('core', 'Subtask')
+    for subtask in Subtask.objects.filter(computation_deadline__isnull=True):
+        subtask.computation_deadline = parse_timestamp_to_utc_datetime(
+            deserialize_message(
+                subtask.task_to_compute.data
+            ).compute_task_def['deadline']
+        )
+        subtask.full_clean()
+        subtask.save()
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('core', '0005_auto_20180717_0908'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='subtask',
+            name='computation_deadline',
+            field=models.DateTimeField(blank=True, null=True),
```

Comment:
I am not sure if I get it right, but in the `Subtask` model this field is neither `blank` nor `null`. Is this setting (`(blank=True, null=True)`) necessary for migration?

Reply:
Migration could not be processed without (blank=True, null=True). This is workaround if we want those fields to be not nullable in model.

Reply:
You can do this without the workaround but you need to tweak the migration a bit. It should be non-nullable in the model but you can make it temporarily nullable in the migration:
- Create the field as nullable
- Perform data migration that fills the field
- Modify the field so that it's no longer nullable.

Reply:
What is this workaround fixing ?

Reply:
I think that the problem is that when you add a new, non-nullable field Django requires a default because otherwise it cannot assign it a value. We don't want to add a default.

Reply:
But on the other hand we can't have mismatch between field definition and database structure.

Reply:
Yeah but the structure is in the process of being changed when you're inside a migration. So if it's nullable at first and then not, it still matches the database (at a given point in time).

Reply:
So the migration for each field should have 3 steps:
1. Create with null and blank.
2. Set desired values.
3. Make non-null and non-blank.